bot/plotting/random_plot_generator.py

```diff
@@ -21,10 +24,12 @@ def random_plot_generator(
             default: None
         provided_number_of_comp: numerical
```

Comment:
can you add an explanation for these inputs? can be a separate issue

Reply:
Yes, I will add some sort of description in another PR, I will open an issue. Actually, all of them were added for the tests, in one go only a certain part of this function is executed (depending on the generated random values), hence it becomes really hard to test this function.

Reply:
I see, might be better to split into deterministic functions that you can test and a master function with the randomization that calls the tested functions?

Reply:
I will try that and see if I can remove these variables